src/transform3d.rs

```diff
@@ -740,6 +742,38 @@ where
             self.transform_point2d(point2(min.x, max.y))?,
         ]))
     }
+
+    /// Returns a 2d box that encompasses the result of transforming the given box by this
+    /// transform, if the transform makes sense for it, or `None` otherwise.
+    pub fn outer_transformed_box2d(&self, b: &Box2D<T, Src>) -> Option<Box2D<T, Dst>>
+    where
+        T: Sub<Output = T> + Div<Output = T> + Zero + PartialOrd,
+    {
+        Some(Box2D::from_points(&[
+            self.transform_point2d(b.min)?,
+            self.transform_point2d(b.max)?,
+            self.transform_point2d(point2(b.max.x, b.min.y))?,
+            self.transform_point2d(point2(b.min.x, b.max.y))?,
+        ]))
+    }
+
+    /// Returns a 3d box that encompasses the result of transforming the given box by this
+    /// transform, if the transform makes sense for it, or `None` otherwise.
+    pub fn outer_transformed_box3d(&self, b: &Box3D<T, Src>) -> Option<Box3D<T, Dst>>
+    where
+        T: Sub<Output = T> + Div<Output = T> + Zero + PartialOrd,
+    {
+        Some(Box3D::from_points(&[
+            self.transform_point3d(b.min)?,
```

Comment:
Can we reorder them so that reviewing is easier? Right now, for example, if I want to see the X component, they go in the following order "min, max, max, min, min,max, min, max". This is an irregular pattern.
Ideally, I'd expect these to go just like digits in binary numbers, i.e.:
X: 01010101
Y: 00110011
Z: 00001111

Reply:
OK I'm not sure I got what you meant exactly but I took your binary analogy with min=0 and max=1 and reordered them lexicographical order (000, 001, 010, 011, ...)